### PR TITLE
Update doxygen (cont'd)

### LINKED ISSUE #1535: Missing doxygen groupings
**Problem description**

While Cantera's C++ core provides detailed information on classes and functions, many classes and functions lack `@ingroup` labels that would generate convenient structures of the doxygen-generated HTML pages, specifically the [Modules tab](https://cantera.org/documentation/docs-2.6/doxygen/html/modules.html).

For others, `@ingroup` are defined, but there is no associated `@defgroup`. Examples are:

- `@ingroup electrochem`
- `@ingroup odeGroup`
- `@ingroup globalUtilFuncs`

Beyond, the generated output may often not be what was intended:

- [x] `@defgroup physConstants` misses most defined constants (proposed resolution in #1534)
- [x] `@defgroup intGroup` and `@defgroup relatedProps` define no methods/classes.
- [x] `@defgroup solvesp_methods` and `@defgroup solvesp_bulkFunc` add no useful information.

In either case, the list is likely incomplete.

As an aside, some of the newer Cantera capabilities do not show up in the Modules drop-down (example: `AnyValue` / `AnyMap`), and some of the groupings could be better structured.

**System information**

- Cantera version: 3.0.0b1

### PR DESCRIPTION
**Pull request by ischoegl**
**Changes proposed in this pull request**

- Restructure doxygen File Input/Output documentation
- Restructure `oneD` documentation
- Restructure `zeroD` documentation
- Add tree view

**If applicable, fill in the issue number this pull request is fixing**

Resolves #1535

**If applicable, provide an example illustrating new features this pull request is introducing**

**Checklist**

- [x] The pull request includes a clear description of this code change
- [x] Commit messages have short titles and reference relevant issues
- [x] Build passes (`scons build` & `scons test`) and unit tests address code coverage
- [x] Style & formatting of contributed code follows [contributing guidelines](https://github.com/Cantera/cantera/blob/main/CONTRIBUTING.md)
- [x] The pull request is ready for review

### CODE REVIEW

File: include/cantera/oneD/Boundary1D.h [hunk 1]
```diff
@@ -395,6 +392,8 @@ class ReactingSurf1D : public Boundary1D
     vector_fp m_fixed_cov;
 };
 
+//! @}
+
```
Comment (speth) [inline include/cantera/oneD/Boundary1D.h]: When the end of the group is hundreds of lines away, the note as to what group is being ended is helpful (as you have in some other cases).

File: include/cantera/oneD/OneDim.h [hunk 2]
```diff
@@ -45,7 +45,8 @@ class OneDim
     //!     superseded by addDomain() using shared_ptr<Domain1D>
     void addDomain(Domain1D* d);
 
-    //! Return a reference to the Jacobian evaluator.
+    //! Return a reference to the Jacobian evaluator of an OneDim object.
+    //! @ingroup derivGroup
```
Comment (speth) [inline include/cantera/oneD/OneDim.h:49]: I see what you're trying to do by adding the class name to the docstring, but the way member functions are shown in Doxygen's "modules" leaves a fair bit to be desired, as I'm sure you've noticed. While I think the two groups touched on in this PR are okay, I'd suggest not going any further in the direction of trying to populate such groups, if you hadn't already come to that conclusion.

File: include/cantera/oneD/StFlow.h [hunk 3]
```diff
@@ -32,10 +32,14 @@ enum offset
...
 
+//! @defgroup flowGroup Flow Domains
+//! One-dimensional flow domains.
+//! @ingroup onedGroup
+
 /**
  *  This class represents 1D flow domains that satisfy the one-dimensional
  *  similarity solution for chemically-reacting, axisymmetric flows.
- *  @ingroup onedim
+ *  @ingroup flowdGroup
```
Comment (speth) [inline include/cantera/oneD/StFlow.h]: Should this be `flowDeviceGroup`?

File: include/cantera/kinetics/solveSP.h [hunk 4]
```diff
@@ -486,5 +489,8 @@ class solveSP
 public:
     int m_ioflag = 0;
 };
+
+//! @}
```
Comment (speth) [inline include/cantera/kinetics/solveSP.h]: I'm not quite sure why, but Doxygen is giving me the warning:
```/Users/speth/src/cantera/include/cantera/kinetics/solveSP.h:497: warning: end of file while inside a group```

**Review by speth — CHANGES_REQUESTED** [review]
Thanks for working on this, @ischoegl. I had just a few small comments.

File: include/cantera/oneD/StFlow.h (hunk 3 above)
Comment (ischoegl) [reply include/cantera/oneD/StFlow.h]: Typo .. should be `flowGroup`.

File: include/cantera/oneD/OneDim.h (hunk 2 above)
Comment (ischoegl) [reply include/cantera/oneD/OneDim.h:49]: I tried to collect some spots where Jacobians are evaluated, as they are relevant to the new `derivGroup` module. I indeed tried to mitigate the lacking presentation. I have no intention to make this more wide-spread.

There is some need to document `PreconditionerBase` etc, which hopefully will take care of some of these stop-gap entries.

File: include/cantera/kinetics/solveSP.h (hunk 4 above)
Comment (ischoegl) [reply include/cantera/kinetics/solveSP.h]: I pointed my doxygen to 1.9.1 (same as the GH runner), and don't get warnings. I am counting 3 instances of `//! @{` and `//! @}`, respectively.

**Comment by ischoegl** [comment]
@speth ... I *believe* I addressed everything.

File: include/cantera/kinetics/solveSP.h (hunk 4 above)
Comment (speth) [reply include/cantera/kinetics/solveSP.h]: Interesting. I'm running 1.9.6 locally, but as you say, the delimiters look correct and balanced, so I'm not sure what's driving this warning. I'm not too concerned about it.

File: include/cantera/oneD/OneDim.h (hunk 2 above)
Comment (speth) [reply include/cantera/oneD/OneDim.h:49]: `breathe` may eventually provide us with an opportunity to adjust the presentation, so I think this is fine for now.

**Review by speth — APPROVED** [review]
Thanks, @ischoegl. This looks good to me.

**Pull request merged**